pynicotine/gtkgui/search.py

```diff
@@ -1452,20 +1448,21 @@ def on_refilter(self, *_args):
             except re.error:
                 filter_out = None
 
+        # Allow the use of seperator pipes | OR spaces & ampersands
         if filter_size:
-            filter_size = filter_size.split("|")
+            filter_size = re.sub(r"[\s&]", '|', filter_size).split("|")
```

Comment:
There is a slight problem where multiple `|` characters appear if more than one seperator is used between conditions and are stored in the presets like such. For example space+"` & `"+space produces "`|||`". Empty conditions are therefore being passed into check_filter()

WIP a regex that can split better.

Reply:
Resolved - Split at | pipes ampersands & space(s) but don't split operators spaced before digit

`(?:[|&\s])+(?<![<>!=]\s)`

![Screenshot_2022-11-05_23-45-39](https://user-images.githubusercontent.com/88614182/200147258-8ba6a276-efd9-45b7-86e1-f7dc10a358b3.png)